--- guile-1.8.8/libguile/Makefile.am	2010-12-13 17:24:39.000000000 +0000
+++ guile-1.8.8/libguile/Makefile.am	2013-12-26 19:22:43.556239446 +0000
@@ -313,26 +313,6 @@
 
 endif
 
-# Stack limit calibration for `make check'.  (For why we do this, see
-# the comments in measure-hwm.scm.)  We're relying here on a couple of
-# bits of Automake magic.
-#
-# 1. The fact that "libguile" comes before "test-suite" in SUBDIRS in
-# our toplevel Makefile.am.  This ensures that the
-# stack-limit-calibration.scm "test" will be run before any of the
-# tests under test-suite.
-#
-# 2. The fact that each test is invoked as $TESTS_ENVIRONMENT $test.
-# This allows us to ensure that the test will be considered to have
-# passed, by using `true' as TESTS_ENVIRONMENT.
-#
-# Why don't we care about the test "actually passing"?  Because the
-# important thing about stack-limit-calibration.scm is just that it is
-# generated in the first place, so that other tests under test-suite
-# can use it.
-TESTS = stack-limit-calibration.scm
-TESTS_ENVIRONMENT = true
-
 stack-limit-calibration.scm: measure-hwm.scm guile$(EXEEXT)
 	$(preinstguile) -s $(srcdir)/measure-hwm.scm > $@
 
